### ISSUE TITLE
Swap ordinals and method params in display names

### ISSUE THREAD

**Issue by adamralph** [adamralph/xbehave.net#233]
Currently:

```
Foo.Bar(baz: "abc") [02.01] Given something
Foo.Bar(baz: "abc") [02.02] When something
Foo.Bar(baz: "abc") [02.03] Then something
Foo.Bar(baz: "xyz") [01.01] Given something
Foo.Bar(baz: "xyz") [01.02] When something
Foo.Bar(baz: "xyz") [01.03] Then something
```

Better?

```
Foo.Bar[01.01](baz: "xyz") Given something
Foo.Bar[01.02](baz: "xyz") When something
Foo.Bar[01.03](baz: "xyz") Then something
Foo.Bar[02.01](baz: "abc") Given something
Foo.Bar[02.02](baz: "abc") When something
Foo.Bar[02.03](baz: "abc") Then something
```

**Comment by ursenzler** [comment]
I like the proposed order better, that the currently implemented one :-)

**Comment by adamralph** [comment]
Something has just occurred to me. In 1.x, there was an undocumented feature which allowed the suppression of argument values in scenario names, which would leave it to the test author to embed the argument values in the step names instead, i.e.

``` C#
[Scenario]
[Example(1, 2)]
[Example(2, 3)]
public void Bar(int x, int y)
{
    "Given {0}"
        .f(() => ...);

    "When something"
        .f(() => ...);

    "Then {1}"
        .f(() => ...);
}
```

```
Foo.Bar [02.01] Given 1
Foo.Bar [02.02] When something
Foo.Bar [02.03] Then 2
Foo.Bar [01.01] Given 2
Foo.Bar [01.02] When something
Foo.Bar [01.03] Then 3
```

But over time, I realized that switching off the argument values in the scenario names was rarely a good idea, even if the values are also embedded in the step names, so I removed the undocumented feature in 2.0.

In the sorting token `[01.01]`, the first number is the scenario invocation ordinal (or example ordinal) and was added to ensure that the steps for a single example appear together in the output. Given that the argument values will always be included in the scenario name in 2.0, that first number is now redundant and can be removed:

```
Foo.Bar(baz: "abc") [01] Given something
Foo.Bar(baz: "abc") [02] When something
Foo.Bar(baz: "abc") [03] Then something
Foo.Bar(baz: "xyz") [01] Given something
Foo.Bar(baz: "xyz") [02] When something
Foo.Bar(baz: "xyz") [03] Then something
```

I can't think of a case where the sorting wouldn't work, unless someone defined two examples with the same values, but that would have no purpose and the duplicate should be removed.

**Comment by adamralph** [comment]
No longer valid, since the scenario number has been removed in 2.0. Only the step number remains, e.g.

```
Foo.Bar(baz: "abc") [01] Given something
Foo.Bar(baz: "abc") [02] When something
Foo.Bar(baz: "abc") [03] Then something
Foo.Bar(baz: "xyz") [01] Given something
Foo.Bar(baz: "xyz") [02] When something
Foo.Bar(baz: "xyz") [03] Then something
```
